core/src/main/java/com/chopshop166/chopshoplib/drive/DiffDriveSubsystem.java

```diff
@@ -0,0 +1,266 @@
+    }
+
+    /** Reset the encoders. */
+    public void resetEncoders() {
+        left.getEncoder().reset();
+        right.getEncoder().reset();
+    }
+
+    /** Reset the gyro. */
+    public void resetGyro() {
+        gyro.reset();
+    }
+
+    /**
+     * Get the turn rate.
+     * 
+     * @return The turn rate in degrees/second.
+     */
+    public double getTurnRate() {
+        return gyro.getRate();
+    }
+
+    /**
+     * Get an object containing wheel speeds.
+     * 
+     * @return Left and right wheel speeds.
+     */
+    public DifferentialDriveWheelSpeeds getWheelSpeeds() {
+        return new DifferentialDriveWheelSpeeds(left.getEncoder().getRate(), right.getEncoder().getRate());
+    }
+
+    /**
+     * Get the average encoder distance.
+     *
+     * @return The average distance.
+     */
+    private double encoderAvg() {
+        return (left.getEncoder().getDistance() + right.getEncoder().getDistance()) / 2;
+    }
+
+    /**
+     * Tank drive using motor controller speeds (RPM).
+     * 
+     * @param left  The left speed.
+     * @param right The right speed.
+     */
+    public void tankDriveSetpoint(final double left, final double right) {
+        this.left.setSetpoint(left);
+        this.right.setSetpoint(right);
+        driveTrain.feed();
+    }
+
+    /**
+     * Drive using controller axes.
+     *
+     * @param forward The forward direction.
+     * @param turn    The direction to turn.
+     * @return A command that will run until interrupted.
+     */
+    public CommandBase drive(final DoubleSupplier forward, final DoubleSupplier turn) {
+        return running("Drive", () -> {
+            final double yAxis = forward.getAsDouble();
+            final double xAxis = turn.getAsDouble();
+            driveTrain.arcadeDrive(yAxis, xAxis);
+        });
+    }
+
+    /**
+     * Drive a given distance at the given speed.
+     * 
+     * @param distance The distance in meters.
+     * @param speed    The speed in motor controller units.
+     * @return The command.
+     */
+    public CommandBase driveDistance(final double distance, final double speed) {
+        return functional("Drive Distance", () -> {
+            resetEncoders();
+        }, () -> {
+            driveTrain.arcadeDrive(speed, 0);
+        }, (interrupted) -> {
+            driveTrain.stopMotor();
+        }, () -> {
+            return encoderAvg() >= distance;
+        });
+    }
+
+    /**
+     * Turn to a given relative angle at the given speed.
+     * 
+     * @param degrees The angle to turn by in degrees.
+     * @param speed   The speed to turn at in motor controller units.
+     * @return The command.
+     */
+    public CommandBase turnDegrees(final double degrees, final double speed) {
+        return functional("Turn Degrees", () -> {
+            resetGyro();
+        }, () -> {
+            double realSpeed = speed;
+            if (degrees < 0 && speed > 0) {
```

Comment:
What if `degrees > 0`, but `speed < 0`? We probably want to make the signs match.

Reply:
"If we're turning left but the speed will turn us to the right"

Maybe we should be comparing the signs of the two?